projects/packages/videopress/src/client/block-editor/blocks/video/components/videopress-uploader/uploader-progress.native.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * External dependencies
+ */
+import { MediaUploadProgress, VIDEO_ASPECT_RATIO } from '@wordpress/block-editor';
+import { Icon } from '@wordpress/components';
+import { usePreferredColorSchemeStyle } from '@wordpress/compose';
+import { View } from 'react-native';
+/**
+ * Internal dependencies
+ */
+import { VideoPressIcon } from '../icons';
+import style from './style.scss';
+
+const UploaderProgress = ( { file, onDone } ) => {
+	const containerStyle = usePreferredColorSchemeStyle(
+		style[ 'videopress-uploader__container' ],
+		style[ 'videopress-uploader__container--dark' ]
+	);
+	const iconStyle = usePreferredColorSchemeStyle(
+		style[ 'videopress-uploader__icon' ],
+		style[ 'videopress-uploader__icon--dark' ]
+	);
+
+	return (
+		<MediaUploadProgress
+			mediaId={ file?.id }
+			onFinishMediaUploadWithSuccess={ payload => {
+				const { metadata, mediaServerId } = payload;
+				onDone( { id: mediaServerId, guid: metadata?.videopressGUID } );
+			} }
+			renderContent={ ( { isUploadFailed, retryMessage } ) => {
+				return (
+					<View style={ [ containerStyle, { aspectRatio: VIDEO_ASPECT_RATIO } ] }>
+						<Icon icon={ VideoPressIcon } { ...iconStyle } />
+						{ isUploadFailed && <Text style={ style.uploadFailedText }>{ retryMessage }</Text> }
```

Comment:
Looks like the import for Text is missing

Reply:
Oh, good catch 🤦. I'll update it as soon as possible, thanks!

Reply:
This made me realize that we haven't covered the upload failure case, which doesn't work as expected. I'll push a commit with a fix and update the testing instructions to cover the case. Sorry for the inconvenience 🙇.

cc @SiobhyB @jhnstn I'll let you know when it's ready in case you could do another review. Thanks!

Reply:
> [..] and update the testing instructions to cover the case. [...]

FYI: I've added the test case "Upload fail" to the **Testing instructions**.